### DISCUSSION TITLE
Nitro Plane on iNav

### DISCUSSION THREAD

**Discussion by Dartanyaun** [iNavFlight/inav#6576] [Q&A]
Has anyone attempted a nitro plane on iNav before? I'm certain there are one hundred good reasons not to, but I'd like to try and pull it off. My goal is to take this fully functioning nitro powered RC plane and make it compatible with iNav.

    There isn't a profile for a nitro so I've had to improvise a standard plane profile. I'm using a crossfire receiver and tango transmitter and have had no issues with them. My problem lies in when I try to set the throttle to a servo and not an ESC like electric aircraft, that seems to really agitate my F405SE flight controller. I'm using servos 5,6,7, and 8 as Aileron, Elevator, Rudder, and Throttle. Whenever I set my throttle to 100% on the sticks and give full aileron deflection, my board stutters terribly and the servos freeze for a second or two. 

     In the air this would be catastrophic so I need to try and solve it on the ground. I flashed the board with the proper firmware so I don't believe the problem lies there. I can try to get any info necessary to the right people if I could have a little help.

Thank you in advance, 
Dartanyaun

**Comment by dronevate** [comment]
i have done it, but had problems with vibration. and then gave up when my plane went facefirst into the ground, full throttle.... anyways, what are you using for power distribution? and as far as throttle goes i just set it up as pwm and ran the s1 signal to the throttle servo. and a ground i believe. ive tried hard to forget that plane... lol but im down to help get an established setup on inav for gas... theyre so fun and powerful. 
my setup was 
tx: jumper t8sg w r9m2019 jr module
jumper r900 mini
jhemcu wing 10? flashed with f35 firmware
on a balsa that i bought on consignment for $75 that looked like a extra 300 with a os 50sx max mounted almost upside down (i guess to help with flooding)

**Reply by Jmay5** [reply]
Hi doenevate,

Im currently working an big gasser Cub that I'm planning on using a Matek F765 FC. Currently I have just added a camera with head tracker. Just to see if the vibrations were going to be too much. To my surprise it worked a hole to better than i thought!! so Im moving forward with this project.

So a couple questions I have for you mind you I haven't received the FC yet I'm just getting an mental picture at how this is going to work.

1) The throttle control. How do you tell the FC not close the throttle to 0 and kill the engine in flight and when the flight is finished how do you kill the engine?

2) You said that posable vibration caused the aircraft to imminently nose dive in to the ground. Curious couldn't you switch to manual mode to save the plane? just trying to understand what may have happened. 

PS. here's a link to my first test flight 

https://www.youtube.com/watch?v=2IR655SJsuQ

**Comment by DecoyBig** [comment]
Hi everyone.

Is there any further success in combining Matek FC with iNav for a nitro / gasoline aircraft? For those who managed to setup this combination in a proper way please advise how it was possible to do step by step.

**Reply by Jmay5** [reply]
So I couldn’t find any real support using INAV so I decided to go to ardupiliot it’s a little more complicated but had the answers I needed. So far so good.

**Reply by DecoyBig** [reply]
Did you connect ESC to Matek FC or leave it without ESC?

**Reply by Jmay5** [reply]
No esc I plugged in a throttle servo where the esc plugs in to. You need the FC to have control over the throttle during flight modes. My problem with INAV I couldn’t figure out to set max and min throttle settings without killing the engine. And then when you want it to
Go to engine kill

**Reply by breadoven** [reply]
Can't you set up the throttle without throttle stop so it always idles at min stick. To kill the motor use a logic condition controlled by something (probably a switch or possibly throttle trim full down) to set OVERRIDE_THROTTLE to a value that stops the motor. 

https://github.com/iNavFlight/inav/blob/master/docs/Programming%20Framework.md

**Reply by Jmay5** [reply]
Maybe but I never got any support from February from my original post so I moved to another platform and it works better than expected.
